File: media-server/MediaServer.Core/Models/UserProfile.cs
using MediaServer.Core.Repositories;
using MediaServer.Models;
using System;

namespace MediaServer.Core.Models
{
    public sealed class UserProfile
    {
        /// <summary>
        /// This user's id in context of the room
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// This user's name in context of the room
        /// </summary>
        public string Username { get; set; }

        /// <summary>
        /// The devices associated with this user
        /// </summary>
        public IRemoteDeviceCollection Devices { get; } = new RemoteDeviceCollection();

        /// <summary>
        /// The room in which this user belongs to
        /// </summary>
        public IRoom Room { get; }

        public UserProfile(IRoom room)
        {
            Room = room ?? throw new ArgumentNullException(nameof(room));
        }

        public override string ToString() => $"[UserProfile {Username}, Id={Id}]";
    }
}

File: media-client/src/models/user-profile.js
export default class UserProfile
{
    get id () { return this._id; }
    set id(value) { this._id = value; }

    get username()  { return this._username; }
    set username(value) { this._username = value; }

    get isOnline() { return this._isOnline; }
    set isOnline(value) { this._isOnline = value; }

    constructor(props)
    {
        this.id = props.id;
        this.username = props.username;
        this.isOnline = props.isOnline;
    }
}